### PR TITLE
[SYCL][NFC] Drop output from FilterSelector E2E tests

### PR DESCRIPTION
**Pull request by AlexeySachkov**
Removed most uses of `std::cout` from `FilterSelector` E2E tests.

There is no need in positive or informational output in those tests: it is not displayed in CI logs when tests pass, it is not checked by tools like `FileCheck`, etc. Expectation is that less I/O will allow tests to pass quicker

### CODE REVIEW

**Comment by maarquitos14** [comment]
Would it be worth it to keep them under a debug flag? Are they useful for debug?

**Comment by AlexeySachkov** [comment]
> Would it be worth it to keep them under a debug flag? Are they useful for debug?

If test prints information about which step failed, then I don't think that information about passed steps is necessary. And I've tried to preserve the former by adding messages to `asserts` we have. Yes, there could be a situations where test crashes and not failing any assertions, so prints would help localize the crash, but so will do a debugger and developer can quickly add necessary prints locally.

For these particular tests, first prints like `std::cout << "ONEAPI_DEVICE_SELECTOR=" << envVal << std::endl;` may be useful to understand how exactly the test is launched, but I believe that the same information is available in logs already, because `lit` prints the command it launched.

Having a debug flag may look like a trade-off, but it may actually be worse: unless we launch tests with that flag (or at least build them with it), there is no guarantee that code hidden under it compiles, making it outdated the moment it was written: developer may have to first fix those debug prints first and only then start debugging.

To summarize: I don't think that those prints are useful enough to preserve them in some form.

**Comment by maarquitos14** [comment]
> > Would it be worth it to keep them under a debug flag? Are they useful for debug?
> 
> If test prints information about which step failed, then I don't think that information about passed steps is necessary. And I've tried to preserve the former by adding messages to `asserts` we have. Yes, there could be a situations where test crashes and not failing any assertions, so prints would help localize the crash, but so will do a debugger and developer can quickly add necessary prints locally.
> 
> For these particular tests, first prints like `std::cout << "ONEAPI_DEVICE_SELECTOR=" << envVal << std::endl;` may be useful to understand how exactly the test is launched, but I believe that the same information is available in logs already, because `lit` prints the command it launched.
> 
> Having a debug flag may look like a trade-off, but it may actually be worse: unless we launch tests with that flag (or at least build them with it), there is no guarantee that code hidden under it compiles, making it outdated the moment it was written: developer may have to first fix those debug prints first and only then start debugging.
> 
> To summarize: I don't think that those prints are useful enough to preserve them in some form.

Understood. Thanks for the explanation.

**Review by maarquitos14 — APPROVED** [review]
LGTM.
